lib/engine/game/g_18_rhl/game.rb

```diff
@@ -1029,7 +1077,73 @@ def check_distance(route, visits)
 
           return super unless visits.find { |v| !RGE_HEXES.include?(v.hex.name) && v.hex.tile.color == :red }
 
-          raise GameError, 'Route for 8 trains cannot include any off-board hexes besides RGE ones'
+          raise GameError, 'Route for 8 trains cannot include any off-board hexes besides the RGE ones'
+        end
+
+        def revenue_for(route, stops)
+          revenue = super
+          corporation = route.train.owner
+
+          [off_board_out_tokened_penalty(route, stops, corporation),
```

Comment:
this line and 1105, maybe turn this into a method so you can reuse in both places

Reply:
Done